src/daemon/auth/common/crypt.rs

```diff
@@ -35,8 +35,14 @@ pub(crate) fn load_or_create_key(key_path: &Path) -> KrillResult<Vec<u8>> {
         openssl::rand::rand_bytes(&mut key_bytes)
             .map_err(|err| Error::Custom(format!("Unable to generate symmetric key: {}", err)))?;
 
-        let mut f = File::create(key_path)?;
-        f.write_all(&key_bytes)?;
+        let mut f = File::create(key_path)
+            .map_err(|e| KrillIoError::new(format!("Could not create key file '{}'", key_path.to_string_lossy()), e))?;
```

Comment:
The same error message is returned by `fn load_key()` above, will it be possible to distinguish the two cases from each other in the logs?

Reply:
I don't see what you mean. I think all errors here are subtly different. E.g. "Could not create key file '..'" vs "Could not write to key file '..'"

Reply:
In `Signer::create_key()` it says:
```
       let mut f = File::create(&path)
            .map_err(|e| KrillIoError::new(format!("Could not create key file '{}'", path.to_string_lossy()), e))?;
```
That is the same message as in this fn. Will the two cases be obviously different in the logs? Is the difference in key path value enough or should the logged message be different?

Reply:
Right, I think it's enough to have the context of the filename and the `io::Error`.